src/PhpSessionPersistence.php

```diff
@@ -105,4 +147,82 @@ private function generateSessionId() : string
     {
         return bin2hex(random_bytes(16));
     }
+
+    /**
+     * Generate cache headers for a given session cache_limiter value.
+     * @param string $cacheLimiter
+     * @param int $cacheExpire
+     */
+    private function generateCacheHeaders(string $cacheLimiter, int $cacheExpire = 0) : array
+    {
+        // Unsupported cache_limiter
+        if (! isset(self::$supported_cache_limiters[$cacheLimiter])) {
+            return [];
+        }
+
+        // cache_limiter: 'nocache'
+        if ('nocache' === $cacheLimiter) {
+            return [
+                'Expires'       => self::CACHE_PAST_DATE,
+                'Cache-Control' => 'no-store, no-cache, must-revalidate',
+                'Pragma'        => 'no-cache',
+            ];
+        }
+
+        $maxAge       = 60 * $cacheExpire;
+        $lastModified = $this->getLastModified($_SERVER['SCRIPT_FILENAME'] ?? '');
+
+        // cache_limiter: 'public'
+        if ('public' === $cacheLimiter) {
+            return [
+                'Expires'       => gmdate(self::HTTP_DATE_FORMAT, time() + $maxAge),
+                'Cache-Control' => sprintf('public, max-age=%d', $maxAge),
+                'Last-Modified' => $lastModified,
+            ];
+        }
+
+        // cache_limiter: 'private'
+        if ('private' === $cacheLimiter) {
+            return [
+                'Expires'       => self::CACHE_PAST_DATE,
+                'Cache-Control' => sprintf('private, max-age=%d', $maxAge),
+                'Last-Modified' => $lastModified,
+            ];
+        }
+
+        // last possible case, cache_limiter = 'private_no_expire'
+        return [
+            'Cache-Control' => sprintf('private, max-age=%d', $maxAge),
+            'Last-Modified' => $lastModified,
+        ];
+    }
+
+    /**
+     * Return the Last-Modified header line based on script name mtime
+     * @return string
+     * @return string|false
+     */
+    private function getLastModified(string $filename)
+    {
+        if ($filename && is_file($filename)) {
+            return gmdate(self::HTTP_DATE_FORMAT, filemtime($filename));
+        }
+
+        return false;
+    }
+
+    /**
+     * Check if the response already carries cache headers
+     * @param ResponseInterface $response
+     * @return bool
+     */
+    private function responseAlreadyHasCacheHeaders(ResponseInterface $response) : bool
+    {
+        return (
+               $response->hasHeader('Expires')
```

Comment:
Reduce the indentation of this line; `||` operators should align with this line.